posts/comparing-java-lts-releases.md

```diff
@@ -0,0 +1,129 @@
+
+# Future Directions and World Domination
+Java has had quite the glow-up in the last 5 years.  Oracle took a language that was on the cusp of becoming one of the "old-school" languages like C or Fortran, and re-made it in the image of languages that are succeeding today for modern use-cases.  The great thing about Java used to be it was easy to learn.  There are no function pointers, there is only one type of casting, no multiple inheritance, you could run Java on any platform, and you do not need to know how to use any command line tools because your IDE takes care of most of it.  If you are a novice, the command line in your Eclipse IDE is how you interact with the applicaiton.  If you were curious, you built a JApplet.  If you were an enterprise developer, you built a REST interface underneath some HTML you copy/pasted off the web.  This worked for awhile, until our data sets got bigger, our databases all became columnar, our front-ends were stand-alone products like Grafana, and we became more interested in our future results over our current results.  How does a language like Java stand a chance in a world where the function of the object outweights the object itself?  The short answer is, Java needed to be reinvented.  Now we have a robust object-oriented language that can easily fill the role of a functional scripting language.  By trimming the fat inherent to every Java application, Java once again becomes an accessible language.  We no longer need 20 lines of import statements, we can just make a module.  We no longer need to compile thousands of lines of code to see if one crucial piece works as expected, we can just use `jshell`.  We no longer have long-winded anonymous classes that define highly specific types (yes I am talking about you Apache HTTP Client), we have lambda expressions and dynamic types.  We no longer have helper classes designed to troll through your data structures for entries that meet specific criteria, we have the Streams API.  Pretty soon, [the iconic POJO may be replaced with a concept called a `record` in Java 12](https://dzone.com/articles/interesting-jdk-12-features-to-watch-out-for).  This initiaive by Oracle to keep Java modern ensures it will be used for many more years to come.  
```

Comment:
```suggestion
Java has had quite the glow-up in the last 5 years.  Oracle took a language that was on the cusp of becoming one of the "old-school" languages like C or Fortran, and re-made it in the image of languages that are succeeding today for modern use-cases.  The great thing about Java used to be it was easy to learn.  There are no function pointers, there is only one type of casting, no multiple inheritance, you could run Java on any platform, and you do not need to know how to use any command line tools because your IDE takes care of most of it.  If you are a novice, the command line in your Eclipse IDE is how you interact with the application.  If you were curious, you built a JApplet.  If you were an enterprise developer, you built a REST interface underneath some HTML you copy/pasted off the web.  This worked for awhile, until our data sets got bigger, our databases all became columnar, our front-ends were stand-alone products like Grafana, and we became more interested in our future results over our current results.  How does a language like Java stand a chance in a world where the function of the object outweighs the object itself?  The short answer is, Java needed to be reinvented.  Now we have a robust object-oriented language that can easily fill the role of a functional scripting language.  By trimming the fat inherent to every Java application, Java once again becomes an accessible language.  We no longer need 20 lines of import statements, we can just make a module.  We no longer need to compile thousands of lines of code to see if one crucial piece works as expected, we can just use `jshell`.  We no longer have long-winded anonymous classes that define highly specific types (yes I am talking about you Apache HTTP Client), we have lambda expressions and dynamic types.  We no longer have helper classes designed to troll through your data structures for entries that meet specific criteria, we have the Streams API.  Pretty soon, [the iconic POJO may be replaced with a concept called a `record` in Java 12](https://dzone.com/articles/interesting-jdk-12-features-to-watch-out-for).  This initiative by Oracle to keep Java modern ensures it will be used for many more years to come.  
```

Reply:
"and dynamic types" -> type inference